src/GradeBook.tsx

```diff
@@ -99,6 +102,15 @@ function GradeBook() {
 
 	const [activeCourse, setCourse] = useState(0);
 
+	const getSemester = async () => {
+		await fetch('/semester/1', {
+			method: 'GET',
+			headers: { userID: 'user1' },
+		}).then((res) => res.json());
```

Comment:
I am assuming this is inadvertent but your function `getSemester` is returning `Promise<void>` (ie kind of just returning nothing). Also a tip, you are awaiting a promise chain, which is kind of mixing promise chains and async/await.

Here's my recommendation:

```suggestion
		const response = await fetch('/semester/1', {
			method: 'GET',
			headers: { userID: 'user1' },
		})
		return response.json();
```

If that doesn't work, (still no data) a good way to check would be to do:

```js
const response = await fetch(...);
const jsonBody = await response.json(); // async call so needs to be awaited
console.log(jsonbody);
return jsonBody;
```